tests/did_wallet/test_did.py

```diff
@@ -0,0 +1,642 @@
+import asyncio
+import time
+
+import pytest
+import clvm
+from src.simulator.simulator_protocol import FarmNewBlockProtocol
+from src.types.peer_info import PeerInfo
+from src.util.ints import uint16, uint32, uint64
+from tests.setup_nodes import setup_simulators_and_wallets
+from src.consensus.block_rewards import calculate_base_fee, calculate_block_reward
+from src.wallet.did_wallet.did_wallet import DIDWallet
+from src.wallet.did_wallet import did_wallet_puzzles
+from clvm_tools import binutils
+from src.types.program import Program
+from src.wallet.derivation_record import DerivationRecord
+from src.types.coin_solution import CoinSolution
+from blspy import AugSchemeMPL
+from src.types.spend_bundle import SpendBundle
+from src.wallet.transaction_record import TransactionRecord
+from src.wallet.derive_keys import master_sk_to_wallet_sk
+
+
+@pytest.fixture(scope="module")
+def event_loop():
+    loop = asyncio.get_event_loop()
+    yield loop
+
+
+class TestWalletSimulator:
+    @pytest.fixture(scope="function")
+    async def wallet_node(self):
+        async for _ in setup_simulators_and_wallets(1, 1, {}):
+            yield _
+
+    @pytest.fixture(scope="function")
+    async def two_wallet_nodes(self):
+        async for _ in setup_simulators_and_wallets(
+            1, 2, {"COINBASE_FREEZE_PERIOD": 0}
+        ):
+            yield _
+
+    @pytest.fixture(scope="function")
+    async def two_wallet_nodes_five_freeze(self):
+        async for _ in setup_simulators_and_wallets(
+            1, 2, {"COINBASE_FREEZE_PERIOD": 5}
+        ):
+            yield _
+
+    @pytest.fixture(scope="function")
+    async def three_sim_two_wallets(self):
+        async for _ in setup_simulators_and_wallets(
+            3, 2, {"COINBASE_FREEZE_PERIOD": 0}
+        ):
+            yield _
+
+    async def time_out_assert(self, timeout: int, function, value, arg=None):
```

Comment:
Is this copied & pasted from elsewhere? Consider refactoring so everyone is using the same implementation.